File: app/src/main/java/vergecurrency/vergewallet/helpers/Notification/NotificationCenter.java
package vergecurrency.vergewallet.helpers.Notification;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;
import android.content.IntentFilter;
import android.os.Bundle;

import androidx.localbroadcastmanager.content.LocalBroadcastManager;


public class NotificationCenter {

	public static void addObserver(Context context, NotificationType notification, BroadcastReceiver responseHandler) {
		LocalBroadcastManager.getInstance(context).registerReceiver(responseHandler, new IntentFilter(notification.name()));
	}

	public static void removeObserver(Context context, BroadcastReceiver responseHandler) {
		LocalBroadcastManager.getInstance(context).unregisterReceiver(responseHandler);
	}

	public static void postNotification(Context context, NotificationType notification, Bundle params) {
		Intent intent = new Intent(notification.name());
		// insert parameters if needed
		intent.putExtra("notification", params);
		LocalBroadcastManager.getInstance(context).sendBroadcast(intent);
	}
}

File: app/src/main/kotlin/vergecurrency/vergewallet/helpers/notification/NotificationCenter.kt
package vergecurrency.vergewallet.helpers.notification

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import android.content.IntentFilter
import android.os.Bundle

import androidx.localbroadcastmanager.content.LocalBroadcastManager


object NotificationCenter {

    fun addObserver(context: Context, notification: NotificationType, responseHandler: BroadcastReceiver) {
        LocalBroadcastManager.getInstance(context).registerReceiver(responseHandler, IntentFilter(notification.name))
    }

    fun removeObserver(context: Context, responseHandler: BroadcastReceiver) {
        LocalBroadcastManager.getInstance(context).unregisterReceiver(responseHandler)
    }

    fun postNotification(context: Context, notification: NotificationType, params: Bundle) {
        val intent = Intent(notification.name)
        // insert parameters if needed
        intent.putExtra("notification", params)
        LocalBroadcastManager.getInstance(context).sendBroadcast(intent)
    }
}